README.md
# Flutter Drawer

Drawer package lets you add a beautiful drawer to your Flutter app.

## Installation

1. Add the latest version of package to your pubspec.yaml (and run'dart pub get'):
```yaml
dependencies:
    flutter_drawer_x: ^0.0.1
```

2. Import the package and use it in your Flutter App.
```dart
import 'package:flutter_drawer_x/flutter_drawer_x.dart';
```

## Example 
These are the properties that you can modify:

 - title
 - gradient (color1 and color2)

 <hr>
 <table>
 <tr>
 <td>

```dart
class FlutterDrawer extends StatelessWidget {  
  const FlutterDrawer({Key? key}) : super(key: key);  
  
  @override  
  Widget build(BuildContext context) {  
    return Scaffold(  
      body: Center(  
        child: const FlutterDrawer(  
          title: 'Implement Flutter Package',  
          color1: Colors.indigo,  
          color2: Colors.white,  
        ),  
      ),  
    );  
  }  
}
```

</td>
<td>
<img  src="https://www.google.com/search?q=github+img+source&rlz=1C1KNTJ_enPK1026PK1026&source=lnms&tbm=isch&sa=X&ved=2ahUKEwiX8-zlt9r6AhWbgf0HHaGsDYYQ_AUoAXoECAEQAw&biw=767&bih=744&dpr=1.25#imgrc=2Kf3Pydv1Sbc0M" alt="">
</td>
</tr>
</table>


CHANGELOG.md
## 0.0.1

* Modify Changes


lib/src/flutter_drawer.dart
import 'package:flutter/material.dart';
import 'package:flutter_drawer/src/utils/flutter_drawer_typedef.dart';

class FlutterDrawer extends StatefulWidget {
  const FlutterDrawer({ 
    Key? key,
    this.color1 = Colors.indigo, 
    this.color2 = Colors.white,
    this.title = 'Implement Flutter package',
    
  }) : super(key: key);

  final Color? color1;
  final Color? color2;
  final String title;
  @override
  State<FlutterDrawer> createState() => _FlutterDrawerState();
}

class _FlutterDrawerState extends State<FlutterDrawer> {
  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: Text(widget.title)),
      drawer: SafeArea(
        child: Drawer(
          child: ListView(
            padding: EdgeInsets.zero,
            children: <Widget>[
              DrawerHeader(
                decoration: BoxDecoration(color: widget.color1),
                child: Column(
                  crossAxisAlignment: CrossAxisAlignment.center,
                  children: [
                    Center(
                      child: CircleAvatar(
                        backgroundColor: widget.color2,
                        child: Text('R',
                                style: TextStyle(
                                  fontSize: 30, color: widget.color1,
                                ),
                              ),
                              minRadius: 40,
                              maxRadius: 40,
                      ),
                    ),
                    const SizedBox(
                        height: 2,
                      ),
                    Text('Ryan', 
                    style: TextStyle(color: widget.color2, fontSize: 20),
                    ),
                    const SizedBox( 
                        height: 2,
                    ),
                    Text('ryan@gmail.com',
                      style: TextStyle(color: widget.color2, fontSize: 15),)
                  ],
                ),
              ),
              ListTile(
                      onTap: () { },
                      leading: Icon(
                        Icons.account_box_rounded,
                        color: widget.color1,
                      ),
                      title: const Text("Profile",
                          style: TextStyle(
                              fontSize: 15,
                              fontFamily: 'PTSerif-Bold',
                              letterSpacing: 0.5)),
                    ),
                    const Divider(),
                    ListTile(
                      onTap: () {},
                      leading: Icon(
                        Icons.chat,
                        color: widget.color1,
                      ),
                      title: const Text("Chat",
                          style: TextStyle(
                              fontSize: 15,
                              fontFamily: 'PTSerif-Bold',
                              letterSpacing: 0.5)),
                    ),
                    const Divider(),
                     ListTile(
                      onTap: () {},
                      leading: Icon(
                        Icons.chat,
                        color: widget.color1,
                      ),
                      title: const Text("Upload Post",
                          style: TextStyle(
                              fontSize: 15,
                              fontFamily: 'PTSerif-Bold',
                              letterSpacing: 0.5)),
                    ),
                    const Divider(),
                    ListTile(
                      onTap: () {},
                      leading: Icon(
                        Icons.chat,
                        color: widget.color1,
                      ),
                      title: const Text("About Us",
                          style: TextStyle(
                              fontSize: 15,
                              fontFamily: 'PTSerif-Bold',
                              letterSpacing: 0.5)),
                    ),
                    const Divider(),
                    ListTile(
                      onTap: () {},
                      leading: Icon(
                        Icons.chat,
                        color: widget.color1,
                      ),
                      title: const Text("Privacy & settings",
                          style: TextStyle(
                              fontSize: 15,
                              fontFamily: 'PTSerif-Bold',
                              letterSpacing: 0.5)),
                    ), 
            ],
          )
        )),
      
    );
  }
}

lib/src/flutter_drawer_x.dart
import 'package:flutter/material.dart';
import 'package:flutter_drawer_x/src/utils/flutter_drawer_typedef.dart';

class FlutterDrawer extends StatefulWidget {
  const FlutterDrawer({ 
    Key? key,
    this.color1 = Colors.indigo, 
    this.color2 = Colors.white,
    this.title = 'Implement Flutter package',
    
  }) : super(key: key);

  final Color? color1;
  final Color? color2;
  final String title;
  @override
  State<FlutterDrawer> createState() => _FlutterDrawerState();
}

class _FlutterDrawerState extends State<FlutterDrawer> {
  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: Text(widget.title)),
      drawer: SafeArea(
        child: Drawer(
          child: ListView(
            padding: EdgeInsets.zero,
            children: <Widget>[
              DrawerHeader(
                decoration: BoxDecoration(color: widget.color1),
                child: Column(
                  crossAxisAlignment: CrossAxisAlignment.center,
                  children: [
                    Center(
                      child: CircleAvatar(
                        backgroundColor: widget.color2,
                        child: Text('R',
                                style: TextStyle(
                                  fontSize: 30, color: widget.color1,
                                ),
                              ),
                              minRadius: 40,
                              maxRadius: 40,
                      ),
                    ),
                    const SizedBox(
                        height: 2,
                      ),
                    Text('Ryan', 
                    style: TextStyle(color: widget.color2, fontSize: 20),
                    ),
                    const SizedBox( 
                        height: 2,
                    ),
                    Text('ryan@gmail.com',
                      style: TextStyle(color: widget.color2, fontSize: 15),)
                  ],
                ),
              ),
              ListTile(
                      onTap: () { },
                      leading: Icon(
                        Icons.account_box_rounded,
                        color: widget.color1,
                      ),
                      title: const Text("Profile",
                          style: TextStyle(
                              fontSize: 15,
                              fontFamily: 'PTSerif-Bold',
                              letterSpacing: 0.5)),
                    ),
                    const Divider(),
                    ListTile(
                      onTap: () {},
                      leading: Icon(
                        Icons.chat,
                        color: widget.color1,
                      ),
                      title: const Text("Chat",
                          style: TextStyle(
                              fontSize: 15,
                              fontFamily: 'PTSerif-Bold',
                              letterSpacing: 0.5)),
                    ),
                    const Divider(),
                     ListTile(
                      onTap: () {},
                      leading: Icon(
                        Icons.chat,
                        color: widget.color1,
                      ),
                      title: const Text("Upload Post",
                          style: TextStyle(
                              fontSize: 15,
                              fontFamily: 'PTSerif-Bold',
                              letterSpacing: 0.5)),
                    ),
                    const Divider(),
                    ListTile(
                      onTap: () {},
                      leading: Icon(
                        Icons.chat,
                        color: widget.color1,
                      ),
                      title: const Text("About Us",
                          style: TextStyle(
                              fontSize: 15,
                              fontFamily: 'PTSerif-Bold',
                              letterSpacing: 0.5)),
                    ),
                    const Divider(),
                    ListTile(
                      onTap: () {},
                      leading: Icon(
                        Icons.chat,
                        color: widget.color1,
                      ),
                      title: const Text("Privacy & settings",
                          style: TextStyle(
                              fontSize: 15,
                              fontFamily: 'PTSerif-Bold',
                              letterSpacing: 0.5)),
                    ), 
            ],
          )
        )),
      
    );
  }
}